src/main/java/Task.java

```diff
@@ -0,0 +1,140 @@
+import java.util.*;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.temporal.ChronoUnit;
+
+public class Task {
+    public String type;
+    public String name;
+    public String status;
+    public String date;
+    public LocalDate processedDate;
+    public String fileDate;
+    public boolean hasProcessedDate = false;
+    public int priority = 0;
+
+    public Task(String type, String name) {
+        this.type = type;
+        this.name = name;
+        this.status = "[✗]";
+    }
+
+    public void setDone() {
+        this.status = "[✓]";
+    }
+
+    public int getStatusBinary() {
+        if (this.status == "[✓]") {
+            return 1;
+        } else {
+            return 0;
+        }
+
+    }
+
+    public String getIcon() {
+        if (this.type.equals("todo")) {
+            return "[T]";
+        } else if (this.type.equals("deadline")) {
+            return "[D]";
+        } else {                  // event
+            return "[E]";
+        }
+    }
+
+    /**
+     * Processes the user input date into a localdate format.
+     *
+     * @return String This returns the formatted form of the localdate.
+     */
+
+    public String getProcessedDate() {
+        String dateFormatted = "";
+        this.date.split("\\s");
+        for (int i = 1; i < this.date.split("\\s").length; i++) {
+            if (i == 1) {
+                dateFormatted += this.date.split("\\s")[i];
+            } else {
+                dateFormatted += " " + this.date.split("\\s")[i];
+            }
+        }
+        LocalDate ld = LocalDate.parse(dateFormatted);
+        this.processedDate = ld;
+        this.hasProcessedDate = true;
+        this.fileDate = ld.format(DateTimeFormatter.ofPattern("MMM d yyyy"));
+        return this.fileDate;
+    }
+
+    public String addProcessedDate(String dateFormatted) {
+        //System.out.println(dateFormatted);
+        int month = 0;
+        int day = Integer.parseInt(dateFormatted.split("\\s")[1]);
+        int year = Integer.parseInt(dateFormatted.split("\\s")[2]);
+        if (dateFormatted.split("\\s")[0].equals("Jan")) {
```

Comment:
You might want to store the date information in another way so that you can convert the saved information back to actual entities more easily, instead of writing a switch-like statement.